JotunnLib/Managers/DungeonManager.cs

```diff
@@ -0,0 +1,387 @@
+            if (prefab == null)
+            {
+                throw new ArgumentException("Cannot be null", nameof(prefab));
+            }
+            if (string.IsNullOrEmpty(themeName))
+            {
+                throw new ArgumentException("Cannot be null or empty", nameof(themeName));
+            }
+
+            DungeonGenerator dg = prefab.GetComponentInChildren<DungeonGenerator>();
+            if (dg == null)
+            {
+                Logger.LogError($"Cannot find DungeonGenerator component in prefab {prefab.name}.");
+                throw new ArgumentException("Prefab must contain a DungeonGenerator component", "prefab");
+            }
+
+            DungeonGeneratorTheme dgt;
+            if (dg.gameObject.TryGetComponent(out dgt))
+            {
+                if (!string.IsNullOrEmpty(dgt.m_themeName) && dgt.m_themeName != themeName)
+                {
+                    Logger.LogWarning($"Overwriting existing theme name {dgt.m_themeName} with {themeName}.");
+                }
+            }
+            else
+            {
+                dgt = dg.gameObject.AddComponent<DungeonGeneratorTheme>();
+            }
+            dgt.m_themeName = themeName;
+            themeList.Add(themeName);
+
+            return true;
+        }
+
+
+        /// <summary>
+        ///     Adds a new environment prefab to be registered when <see cref="ZoneSystem.SetupLocations"/> runs.<br />
+        ///     If you intend to use a custom interior environment <see cref="Location.m_interiorEnvironment"/>, this method enables you 
+        ///     to provide a prefab with an appropriately configured <see cref="LocationList"/> containing atleast one
+        ///     <see cref="EnvSetup"/> within <see cref="LocationList.m_environments"/>.
+        /// </summary>
+        /// <param name="assetBundle">The <see cref="AssetBundle"/> containing the prefab.</param>
+        /// <param name="prefabName">The name of the prefab to register.</param>
+        public void RegisterEnvironment(AssetBundle assetBundle, string prefabName)
+        {
+            if (assetBundle == null)
+            {
+                throw new ArgumentException("Cannot be null", nameof(assetBundle));
+            }
+            if (string.IsNullOrEmpty(prefabName))
+            {
+                throw new ArgumentException("Cannot be null or empty", nameof(prefabName));
+            }
+            envNames.Add(prefabName, assetBundle);
+        }
+
+        /// <summary>
+        ///     Adds a new prefab to be registered when <see cref="ZoneSystem.SetupLocations"/> runs.<br />
+        ///     If you intend to use a custom door, this method enables you provide that prefab here.
+        /// </summary>
+        /// <param name="assetBundle">The <see cref="AssetBundle"/> containing the prefab.</param>
+        /// <param name="prefabName">The name of the prefab to register.</param>
+        public void RegisterPrefab(AssetBundle assetBundle, string prefabName)
```

Comment:
What's the reason to have a dedicated method here instead of using `PrefabManager.Instance.AddPrefab` directly? Especially if the prefab has a ZNetView, `ZoneSystem.SetupLocations` might be too late

Reply:
This was originally written before the new asset system, and if dungeon prefabs used here used mocks of materials/sounds/etc from vanilla dungeon parts, they would fail to resolve when using PrefabManager.Instance.AddPrefab.

In general, it's meant to register prefabs that are specifically used by or referenced from the dungeon generator.

The point about znetview is interesting.

I'll dig into this at some point soon - if mocks referencing dungeon resources now work due to the new asset system, this may not be needed at all anymore.

Reply:
I tested out using regular ol `PrefabManager.Instance.AddPrefab` and it seems to work for my purposes now.  Iirc, the issue I had prior to 0.217.46 was that mocks for dungeon resources would not resolve.  I'll be removing the RegisterPrefab method + logic in my next PR update.

Reply:
Addressed in https://github.com/Valheim-Modding/Jotunn/pull/434/commits/f515abe679c8ca82112bb2ff4c435246aca2ed8d